### DISCUSSION TITLE
Test class name changed - Allure overview shows the previous name and the new one as different assets

### DISCUSSION THREAD

**Discussion by xescuder** [allure-framework/allure2#1568] [Questions & Support] [answered]
The overview and behaviour is used by our POs and stakeholders to check coverage.
We're using history results with JUnit5 and Allure. 

If we change a test class name, the overview/behaviour show the previous name and the new one as different test cases, so the coverage is also incorrect.

How are you dealing with this problem?

Thanks in advance.

**Comment by baev** [comment, marked as answer]
So far Allure Report only knows how to detect different results and retries.

 Allure TestOps (our paid tool) is designed differently and it tracks test cases as well (so you know what are the same test cases and what are different ones). And for problems like you mentioned (for example changes of test method signature) we have `AllureId` annotation that can be used to link result to specific test case in Allure TestOps. 

And we also provide a plugin for Intellij IDEA that will automatically apply `AllureId` annotation for every test method you have so we can lower the chances of mistake.

BTW you can achieve the same in Allure Report:

1. Create a custom `LabelAnnotation`, for example `@MyCustomTestCaseId`
2. Annotate every test method with it, so every result will have a label, e.g. `myCustomTestCaseId`
    ```java
    @MyCustomTestCaseId("globally-unique-value")
    void myTest() {
    }
    ```
3. Then implement `io.qameta.allure.listener.TestLifecycleListener#afterTestStop` and override `historyId` for every result that has `myCustomTestCaseId` label.
